pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -134,26 +134,32 @@ def load_state_dict(self, state_dict):
         self.best_score = state_dict['best_score']
         self.patience = state_dict['patience']
 
-    def on_sanity_check_end(self, trainer, pl_module):
-        logs = trainer.callback_metrics
-        self._validate_condition_metric(logs)
-
     def on_validation_end(self, trainer, pl_module):
         self._run_early_stopping_check(trainer, pl_module)
 
+    def on_validation_epoch_end(self, trainer, pl_module):
```

Comment:
since when do we have two of these hooks? what is the difference?
I suspect on_validation_epoch_end end is never actually called. I can only find the other one
also there are no tests in test_callbacks.py for this.

Reply:
I would use just one and if we assume that `on_validation_epoch_end` is a better name, deprecate the other one

Reply:
well `validation_end` is deprecated so I assume `on_validation_end` shall be too @williamFalcon

Reply:
yes exactly.

we can deprecate in the new PR. (these were the old names which we deprecated)

Reply:
lest add deprecation warning as soon as possible so users get enough time...